api/v1alpha1/contentlibraryprovider_types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ContentLibraryProviderSpec defines the desired state of ContentLibraryProvider
+type ContentLibraryProviderSpec struct {
+	// Name describes the name of a content library. This is not necessarily the same as vCenter
+	// name of the content library.
+	Name string `json:"name,omitempty"`
```

Comment:
This field is meant to represent the vSphere content library name. What do we think about making this field optional since we only need the UUID to uniquely identify a vSphere content library.

Reply:
Yeah, optional seems right since the lookup is done by UUID, not by Name. 

Also, how about `OverrideName` (or something like that) to be more clear that this is to simply use a different name for the CL rather than the vSphere set one.

Reply:
How about we call it `vSphereName` or `vSphereLibraryName` (since `vSphereCLName` has abbreviations and `vSphereContentLibraryName` is too long)

Reply:
Removed it since it was getting confusing to have two names in the kubectl output.